packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -200,6 +207,42 @@ export class RemoteMessageProcessor {
 		}
 		assert(batchMetadataFlag !== true, 0x9d6 /* Unexpected batch start marker */);
 
+		// Validate that the sequence numbers are contiguous. If there are non-runtime messages in the batch,
```

Comment:
Am I correct that we get here only for "legacy" batches?
process() that calls this function has separate code path for isGroupedBatch().

Might be worth updating comments, as it's a bit hard to follow code (and all the combinations).

Reply:
Yes, that is true. We have too many combinations :-). I realized that this logic is only for non-grouped ops. There is no way that batches for grouped ops could be incorrect so that's why there is no validation and I kept it that way.

I will add comments to make it clear.